projects/gateway2/translator/plugins/httplisteneroptions/query/query.go

```
@@ -0,0 +1,114 @@
+package query
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	solokubev1 "github.com/solo-io/gloo/projects/gateway/pkg/api/v1/kube/apis/gateway.solo.io/v1"
+	"github.com/solo-io/gloo/projects/gateway2/translator/plugins/utils"
+	"github.com/solo-io/go-utils/contextutils"
+	skv2corev1 "github.com/solo-io/skv2/pkg/api/core.skv2.solo.io/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/types"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	gwv1 "sigs.k8s.io/gateway-api/apis/v1"
+)
+
+type HttpListenerOptionQueries interface {
+	// GetAttachedHttpListenerOptions returns a slice of HttpListenerOption resources attached to a gateway on which
+	// the listener resides and have either targeted the listener with section name or omitted section name.
+	// The returned HttpListenerOption list is sorted by specificity in the order of
+	//
+	// - older with section name
+	//
+	// - newer with section name
+	//
+	// - older without section name
+	//
+	// - newer without section name
+	//
+	// Note that currently, only HttpListenerOptions in the same namespace as the Gateway can be attached.
```

What is the behavior if a user tries to attach a n HttpListenerOptions to a Gateway in a different namespace? How will they know they have misconfigured something?

---

Unfortunately it will fail silently. The signal will be it is not attached. This is applicable to all policy APIs at this point.
We had an issue I thought to track this on the status of the policy, but that will only help for policies that have status, i.e. RouteOption & VHostOption